thunder/core/transforms.py

```diff
@@ -3148,6 +3148,9 @@ def recompute_saved_for_backward(fwd_trace: Trace, bwd_trace: Trace) -> tuple[Tr
 
     producers = find_producer_symbols(fwd_trace, tuple(unvariableify(i) for i in rematerializable), fwd_trace.args)
 
+    trace_tok = set_tracectx(bwd_trace)
```

Comment:
Please set and reset traces only with "try: finally:" blocks. If there's any error between the calls, the trace will not be reset.

Reply:
Why do you set the input `bwd_trace` as the active trace? There are no Thunder operations calls between set and reset, and the input trace shouldn't be modified.

Reply:
would we not use `with tracectx(bwd_trace)`?

Reply:
For this I've taken inspiration from the code in the torch_autograd executor, in particular these lines explain why the need to set the trace context: https://github.com/Lightning-AI/lightning-thunder/blob/3390c922cfbe1b70c42118dfa8aa71adb3bec692/thunder/executors/torch_autograd.py#L33-L40 

@IvanYashchuk Would an acceptable workaround be to create a new empty trace and use it as ctx?

Reply:
Yes, or allow creating Proxies with any name without active tracectx. Maybe all is needed is to return True if trc is None in this function https://github.com/Lightning-AI/lightning-thunder/blob/3390c922cfbe1b70c42118dfa8aa71adb3bec692/thunder/core/proxies.py#L75-L82